### PR TITLE
RPLUS-7308 | Allow to_csv to receive a csv output stream

### PR DESCRIPTION
**Pull request by JonathanWeids**
If it is provided, open a raw streaming connection to the DB and write to it instead of returning a csv array.

### CODE REVIEW

File: lib/data_table.rb [hunk 1]
```diff
@@ -977,6 +943,58 @@ def parameter_binding_character
...
+        header = []
+        rows = []
+
+        if @options.fetch(:write_headers, true)
+          header << columns.map(&:caption).join(',')
+        end
+
+        resultset, _ = get_resultset_and_count(params, options)
+
+        resultset.each do |instance|
```
Comment (JonathanWeids) [inline lib/data_table.rb:988]: This materializes the view. Need to use something like:

`copy_connection.copy_data "/*NO LOAD BALANCE*/ COPY (#{sql}) TO STDOUT WITH (#{options_string})", decoder do
      while line = copy_connection.get_copy_data
        # replacing non-UTF-8 chars with '??'
        temp_file.write line.join(delimiter).encode(encoding, :replace => '??', :invalid => :replace, :undef => :replace)
        temp_file.write "\n"
      end
    end`

Comment (JonathanWeids) [reply lib/data_table.rb:988]: (found in csv_write.rb)

File: lib/data_table.rb [hunk 2]
```diff
@@ -977,6 +943,60 @@ def parameter_binding_character
...
+
+      def find_user_column(column_name, options)
+        options[:columns_layout]['columns'].detect do |c|
+          split_layout_name = c['dataField'].split('.')
+          name = split_layout_name.length > 2 ? split_layout_name.last(2).map(&:to_sym).to_s : split_layout_name.last.to_s
+          column_name.to_s == name
+        end || { 'visible' => false }
+      end
+
+      def generate_csv_output_stream(csv_output_stream, columns)
```
Comment (ablignaut) [inline lib/data_table.rb:965]: @nbrien I guess you were talking about this method?

Either the gem needs to implement the SQL and PG version and check the connection type for tyhe implementation?
Or it should be an abstracto method and each applicatyion implements their own streaming (This is less re-usable though from multiple projects).

@WillemOpperman what is your input here?

Comment (WillemOpperman) [reply lib/data_table.rb:965]: I agree. This will not work for RC.

This code should be in robusta/rc-ui.

**Comment by nbrien** [comment]
We can close this. We wrote an extension in RP.

**Pull request closed without merging**